storm-core/src/jvm/org/apache/storm/cluster/ZKStateStorage.java

```diff
@@ -189,7 +190,15 @@ public void set_data(String path, byte[] data, List<ACL> acls) {
             Zookeeper.setData(zkWriter, path, data);
         } else {
             Zookeeper.mkdirs(zkWriter, Zookeeper.parentPath(path), acls);
-            Zookeeper.createNode(zkWriter, path, data, CreateMode.PERSISTENT, acls);
+            try {
+                Zookeeper.createNode(zkWriter, path, data, CreateMode.PERSISTENT, acls);
+            } catch (RuntimeException e) {
```

Comment:
Not sure how? createNode() wraps it in RuntimeException, so it has to be unwrapped first, right? https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java#L129